### ISSUE TITLE
:sparkles: Create an OTP docker container to use with the simulation code

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#534]
It would be good if we could have an OTP docker container and launch it to both:
- test the integration, and
- actually generate the trips in one easy swoop.

This issue documents the design decisions related to this feature

**Comment by shankari** [comment]
The URL for the OTP server is currently stored in `emission/net/ext_service/otp/planner.json`. This is clearly terrible since it mixes code and config. There are two possible fixes:
- move it to `conf/net/ext_service`, or
- since it only has one variable, use an environment variable `OTP_SERVER` instead

Using an environment variable makes it easier to use in a `docker-compose` file since we can create an internal network and then use the name of the container directly.

**Comment by shankari** [comment]
We also need to decide which docker image to use.
@alvinalexander created a docker container for the SF bay area, which he published at 
https://hub.docker.com/r/alvinghouas/otp-sfbay

This currently has 23 downloads and 1 star.
However, it is also 800MB+ and I can't run it either on my laptop or on my local linux server.
Trying on one of the millennium servers...

**Comment by alvinalexander** [comment]
@shankari If I remember correctly, the large size is because all the map-data was included in the build. A better approach would probably be to create an image that can fetch map data from an external source (such as an s3 bucket)

**Comment by shankari** [comment]
@alvinalexander good point! However, the problem is not just the disk space, but the memory required. On my laptop, the docker image launches, but then the java process is killed.

>  Trying on one of the millennium servers...

This works. I'm going to go with this right now to unblock @njriasan

**Comment by njriasan** [comment]
I just wanted to update this with some details on the memory required. I was able to launch the OTP server on my laptop and in the process it consumed 8.4 GB of RAM. It was also much faster than 10 minutes for me to start up, probably closer to 2.

**Comment by shankari** [comment]
I guess I need a better laptop 😄

I am surprised that it took less than 10 minutes because the millennium machines are fairly beefy. I wonder if this is an I/O issue - maybe I don't have it on the local storage...
